harness/determined/cli/resource_pool.py

```diff
@@ -77,6 +80,43 @@ def list_workspaces(args: Namespace) -> None:
     return
 
 
+@authentication.required
+def list_resource_pools(args: Namespace) -> None:
+    session = setup_session(args)
```

Comment:
"Migrate CLI to SDK" is a goal for ml-sys, but not one that's been investigated all that deeply. If you find session stuff to be too awkward to do with either `determined.common.experimental.determined.Determined` or `determined.experimental.client`, then please tell me about it and we'll figure out how to make it all more reasonable.

Reply:
I feel it's awkward. The `Determined` class needs a password to authenticate. `setup_session()` uses  `authentication.cli_auth`. `authentication.cli_auth` is initialized by the function decorator `@authentication.required`.
```
class Determined:
    def __init__(
        self,
        master: Optional[str] = None,
        user: Optional[str] = None,
        password: Optional[str] = None,
        cert_path: Optional[str] = None,
        cert_name: Optional[str] = None,
        noverify: bool = False,
    ):
        self._master = master or util.get_default_master_address()

        cert = certs.default_load(
            master_url=self._master,
            explicit_path=cert_path,
            explicit_cert_name=cert_name,
            explicit_noverify=noverify,
        )

        auth = authentication.Authentication(self._master, user, password, cert=cert)
        retry = api.default_retry()
        self._session = api.Session(self._master, user, auth, cert, retry)
```

```
def setup_session(args: argparse.Namespace) -> api.Session:
    master_url = args.master or util.get_default_master_address()
    cert = certs.default_load(master_url)
    retry = api.default_retry()

    return api.Session(master_url, args.user, authentication.cli_auth, cert, retry)
```

Reply:
Point taken. This deserves a mass migration, not a piecemeal strategy. I'll withdraw my recommendation for generating a session via the SDK, and I support the decision you've made. Someday we'll find a better solution and we'll apply it everywhere.